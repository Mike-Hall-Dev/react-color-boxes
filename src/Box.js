import React from "react";

function Box({
    id,
    width = 7,
    height = 7,
    backgroundColor = "green",
    handleDelete
}) {
    const deleteBox = () => handleDelete(id)
    return (
        <>
            <div style={{
                width: `${width}rem`,
                height: `${height}rem`,
                backgroundColor
            }}>
                <button onClick={deleteBox}>X</button>
            </div>
        </>
    )
}

export default Box