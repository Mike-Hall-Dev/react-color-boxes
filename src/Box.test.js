import React from "react"
import { render } from "@testing-library/react";
import Box from "./Box"

it("renders without crashing", function () {
    render(<Box />);
});

it("matches the current stored snapshot", function () {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
})