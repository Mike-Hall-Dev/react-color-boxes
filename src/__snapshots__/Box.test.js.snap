// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the current stored snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 7rem; height: 7rem; background-color: green;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
